infrastructure/main.tf

```diff
@@ -0,0 +1,145 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 5.0"
+    }
+  }
+
+  required_version = ">= 1.0.0"
+}
+
+provider "aws" {
+  shared_credentials_files = ["~/.aws/credentials"]
```

Comment:
We don't want to specify here the shared_credentials_files.
You just login via SSO using your profile and terraform should know how to authenticate to AWS.